### PR TITLE
Add pod status info log for e2e creating pods failure

### PR DESCRIPTION
**Pull request by 249043822**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
add pod status logs while e2e tests creating pods, may help identify test problems
#### Which issue(s) this PR fixes:

Fixes #https://github.com/kubernetes/kubernetes/issues/107804#issuecomment-1055184228

#### Special notes for your reviewer:
/cc @SergeyKanzhelev @ehashman 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/framework/pod/resource.go [hunk 1]
```diff
@@ -193,7 +193,7 @@ func podRunningAndReady(c clientset.Interface, podName, namespace string) wait.C
 		}
 		switch pod.Status.Phase {
 		case v1.PodFailed, v1.PodSucceeded:
-			e2elog.Logf("The status of Pod %s is %s which is unexpected", podName, pod.Status.Phase)
```
Comment (SergeyKanzhelev) [inline test/e2e/framework/pod/resource.go:196]: should this also be added at line 202?

**Review by SergeyKanzhelev — COMMENTED** [review]
@249043822 what information do you expect to see in pod status?

File: test/e2e/framework/pod/resource.go (hunk 1 above)
Comment (249043822) [reply test/e2e/framework/pod/resource.go:196]: I think adding to line 202 may print more useless logs, only printing the details while pod becomes terminal status could be enough, right?

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Comment by oomichi** [comment]
Nice addition.

/lgtm
/approve

**Pull request merged**